### PR TITLE
Humane test optput

### PR DESCRIPTION
**Pull request by nenadalm**

### CODE REVIEW

File: src/jx/reporter/karma.cljs [hunk 1]
```diff
@@ -62,6 +63,7 @@
     (karma-result! result)))
 
 (defmethod cljs.test/report [::karma :fail] [m]
+  (println m)
```
Comment (nenadalm) [inline src/jx/reporter/karma.cljs]: this seems to contain correct map now (https://github.com/honzabrecka/karma-reporter/pull/8#issuecomment-262061641)

**Comment by nenadalm** [comment]
This pr just contains initial work - I'll probably continue on the weekend

**Comment by honzabrecka** [comment]
Great! Thank you. I'm just working on diff prototype https://github.com/honzabrecka/karma-reporter/tree/diff

You can try with:

```clojure
(deftest failing-test
  (is (= (inc 1) (inc 2) (inc 3)))
  (is (= {:x [4] :a [1 3]}
         {:a [1 2] :y [6] :x [4]}))
  (is (true? (= true false)) "Whoops!")
  (is (= (range 30) (range 1 50)) (str "Let me add in something\nfancy here: " {:a 1234}))
  (is (> 1 2)))
```

**Comment by nenadalm** [comment]
so... now it works, except the output is not very nice

```
$ ./node_modules/.bin/karma start config/run-all.conf.js
26 11 2016 09:45:47.118:WARN [web-server]: 404: /test/cljs_deps.js
..
LOG: '
FAIL in (map-test) (pjstadig.macro.do_report.cljs$core$IFn$_invoke$arity$variadic@http:57:8)'
LOG: ''
LOG: 'expected: {:a "a", :b "b"}
  actual: {:a "b", :b "b"}
'
LOG: '    diff: - {:a "a"}
          + {:a "b"}
'
Firefox 50.0.0 (Fedora 0.0.0) foo.bar-test map-test FAILED
	
Firefox 50.0.0 (Fedora 0.0.0): Executed 3 of 3 (1 FAILED) (0.209 secs / 0.01 secs)
```

because report- function doesn't return string, but does printing... I'll check what to do about this (probably pr to https://github.com/pjstadig/humane-test-output where report- will call some function which will return string and then it'll print it - we could than use the function returning the string instead of calling report-)

**Comment by nenadalm** [comment]
it seems that with-out-str did the trick (I am not very experienced with clojure[script]). Is it mergeable now?

**Comment by honzabrecka** [comment]
Thank you for your work, but I decided to came up with custom output/diff format.

**Pull request closed without merging**